### PR TITLE
add randomUint256 precompile

### PR DESCRIPTION
**Pull request by zeroXbrock**
## 📝 Summary

adds a new precompile `randomUint256` to generate a random number

## 📚 References

---

* [x] I have seen and agree to CONTRIBUTING.md

### CODE REVIEW

File: core/vm/contracts_suave.go [hunk 1]
```diff
@@ -38,6 +40,16 @@ func (b *suaveRuntime) confidentialInputs() ([]byte, error) {
 	return b.contextGet("confidentialInputs")
 }
 
+func (b *suaveRuntime) randomUint256() (*big.Int, error) {
+	maxU256 := new(big.Int)
+	maxU256.SetString("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF", 16)
```
Comment (lthibault) [inline core/vm/contracts_suave.go:45]: Why are we 1-ing out the U256?  It'll be zeroed out by default, so this feels unnecessary.

**Review by lthibault — APPROVED** [review]
Approving for expediency, but please see comment inline.

File: core/vm/contracts_suave.go (hunk 1 above)
Comment (zeroXbrock) [reply core/vm/contracts_suave.go:45]: Can you explain how it gets zeroed out? I don't see where this happens. My understanding is that this defines the maximum value for the random number generator on the next line

Comment (lthibault) [reply core/vm/contracts_suave.go:45]: ☝️ Disregard.  This is setting the max random value, which is absolutely reasonable.  Resolving.

**Pull request merged**